File: 1888-find-nearest-point-that-has-the-same-x-or-y-coordinate/1888-find-nearest-point-that-has-the-same-x-or-y-coordinate.py
class Solution:
    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:
        idx = -1
        distance = 14143
        for i in range(len(points)):
            if points[i][0] == x or points[i][1] == y:
                d = abs(points[i][0] - x) + abs(points[i][1] - y)
                if d < distance:
                    distance = d
                    idx = i
        return idx

File: 1888-find-nearest-point-that-has-the-same-x-or-y-coordinate/1888-find-nearest-point-that-has-the-same-x-or-y-coordinate.java
class Solution {
    public int nearestValidPoint(int x, int y, int[][] points) {
        int idx = -1, distance = 14143;
        for (int i = 0; i < points.length; i++) {
            if (points[i][0] == x || points[i][1] == y) {
                int d = Math.abs(points[i][0] - x) + Math.abs(points[i][1] - y);
                if (d < distance) {
                    distance = d;
                    idx = i;
                }
            }
        }
        return idx;
    }
}